lib/shared-method.js

```diff
@@ -269,35 +302,55 @@ function coerceAccepts(uarg, desc) {
     throw new badArgumentError(name + ' must be a number');
   }
 
-  // convert strings
-  if (actualType === 'string' && targetType !== 'any' && actualType !== targetType) {
-    // JSON.parse can throw, so catch this error.
-    try {
-      uarg = convertValueToTargetType(uarg, targetType);
-    } catch (e) {
-      var message = util.format('invalid value for argument \'%s\' of type ' +
-        '\'%s\': %s. Received type was %s. Error: %s',
-        name, targetType, uarg, typeof uarg, e.message);
-      throw new badArgumentError(message);
-    }
-  }
   return uarg;
 }
 
-function convertValueToTargetType(value, targetType) {
+function convertToBasicRemotingType(type) {
+  if (Array.isArray(type)) {
+    return type.map(convertToBasicRemotingType);
+  }
+
+  if (typeof type === 'object') {
+    type = type.modelName || type.name;
+  }
+
+  type = String(type).toLowerCase();
+  switch (type) {
+    case 'string':
+    case 'number':
+    case 'date':
+    case 'boolean':
+    case 'buffer':
+    case 'object':
+    case 'any':
+      return type;
+    default:
+      // custom types like MyModel
+      return 'object';
```

Comment:
Here is the scenario I have in mind... Actually I don't think this is an issue. Can you confirm?

```js
Person.prototype.greet = function(msg, someone, cb) {
  this.log('Greetings! - ' + msg);
  someone.greet('hello ' + this.name);
  cb();
}

Person.remoteMethod('greet', {
  isStatic: false,
  accepts: [{type: 'string'}, {type: 'Person'}]
});

Person.prototype.log = function(msg) {
  console.log(this.name + ' says: ' + msg);
}
```